texparser/tar_extractor.py

```diff
@@ -13,28 +15,34 @@ def extract_folder(self, folder, func):
         filenames = os.listdir(folder)
         for filename in filenames:
             try:
-                processes.append(func(folder + "/" + filename))
-            except:
+                path = Path(folder) / filename
+                filename, file_extension = os.path.splitext(path)
+                processes.append(func(filename + file_extension))
+            except FileExistsError:
+                sys.stderr.write("Error message: File already exists. \n")
                 pass  # How should we handle already existing folders?
         return processes
-
     def extract_file_into_folder(self, file_name: str, params: str, ending: str) -> tuple:
         tar_process = any
         if file_name.endswith(ending):
             if ending in file_name:
-                file_folder_path = self.__create_file_folder_path(
+                file_folder_path = self.create_file_folder_path(
                     file_name, ending)
                 process = self.__create_tar_process(
                     params, file_name, file_folder_path)
                 return self.__analyse_process(process)
         return (0, 0)
 
-    def __create_file_folder_path(self, file_name: str, ending: str) -> str:
+    def create_file_folder_path(self, file_name: str, ending: str) -> str:
         file_folder_path = file_name.replace(ending, "")
```

Comment:
instead of working with strings we could think about using paths not a must but might make the code cleaner and less error prone